### PR TITLE
improve names of side effect builders

### PR DESCRIPTION
**Pull request by gabrielittner**
Simplifies and aligns the names to match the DSL methods that they are used for.

### CODE REVIEW

File: flowredux/src/commonMain/kotlin/com/freeletics/flowredux/sideeffects/OnEnterStateMachine.kt [hunk 1]
```diff
@@ -20,7 +20,7 @@ import kotlinx.coroutines.flow.onStart
 import kotlinx.coroutines.launch
 
 @ExperimentalCoroutinesApi
-internal class StartStatemachineOnEnterSideEffectBuilder<SubStateMachineState : Any, SubStateMachineAction : Any, InputState : S, S : Any, A>(
+internal class OnEnterStateMachine<SubStateMachineState : Any, SubStateMachineAction : Any, InputState : S, S : Any, A>(
```
Comment (sockeqwe) [inline flowredux/src/commonMain/kotlin/com/freeletics/flowredux/sideeffects/OnEnterStateMachine.kt:23]: Just personal preference: `OnEnterStartStateMachine` (so add the `Start`). Your call, just a personal preference.

File: flowredux/src/commonMain/kotlin/com/freeletics/flowredux/sideeffects/OnActionStateMachine.kt [hunk 2]
```diff
@@ -19,7 +19,7 @@ import kotlinx.coroutines.launch
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
 
-internal class StartStateMachineOnActionInStateSideEffectBuilder<SubStateMachineState : Any, SubStateMachineAction : Any, InputState : S, ActionThatTriggeredStartingStateMachine : A, S : Any, A : Any>(
+internal class OnActionStateMachine<SubStateMachineState : Any, SubStateMachineAction : Any, InputState : S, ActionThatTriggeredStartingStateMachine : A, S : Any, A : Any>(
```
Comment (sockeqwe) [inline flowredux/src/commonMain/kotlin/com/freeletics/flowredux/sideeffects/OnActionStateMachine.kt:22]: Just personal preference: `OnActionStartStateMachine ` (so add the `Start`). Your call, just a personal preference.

**Review by sockeqwe — APPROVED** [review]

File: flowredux/src/commonMain/kotlin/com/freeletics/flowredux/sideeffects/OnEnterStateMachine.kt (hunk 1 above)
Comment (gabrielittner) [reply flowredux/src/commonMain/kotlin/com/freeletics/flowredux/sideeffects/OnEnterStateMachine.kt]: Makes sense [d2c9d88](https://github.com/freeletics/FlowRedux/pull/505/commits/d2c9d88aeafd019b88f8859559d1b73a052c849a)

File: flowredux/src/commonMain/kotlin/com/freeletics/flowredux/sideeffects/OnActionStateMachine.kt (hunk 2 above)
Comment (gabrielittner) [reply flowredux/src/commonMain/kotlin/com/freeletics/flowredux/sideeffects/OnActionStateMachine.kt]: [d2c9d88](https://github.com/freeletics/FlowRedux/pull/505/commits/d2c9d88aeafd019b88f8859559d1b73a052c849a)

**Pull request merged**